File: rplugin/python3/deoplete/source/vsnip.py
import json
from deoplete.source.base import Base

class Source(Base):
    def __init__(self, vim):
        Base.__init__(self, vim)

        self.name = 'vsnip'
        self.mark = '[vsnip]'
        self.rank = 1000
        self.input_pattern = r'\w\+$'
        self.min_pattern_length = 1
        self.vars = {}

    def gather_candidates(self, context):
        return self.vim.call('vsnip#get_complete_items', context['bufnr'])

File: denops/ddc-sources/vsnip.ts
import {
  BaseSource,
  Candidate,
} from "https://deno.land/x/ddc_vim@v0.5.0/types.ts";
import { Denops, fn } from "https://deno.land/x/ddc_vim@v0.5.0/deps.ts#^";

export class Source extends BaseSource {
  async gatherCandidates(args: {
    denops: Denops;
  }): Promise<Candidate[]> {
    return args.denops.call(
      "vsnip#get_complete_items",
      await fn.bufnr(args.denops),
    ) as Promise<Candidate[]>;
  }
}
